src/enum_variants.rs

```diff
@@ -31,48 +32,63 @@ fn var2str(var: &Variant) -> InternedString {
     var.node.name.name.as_str()
 }
 
-fn partial_match(left: &str, right: &str) -> usize {
-    left.chars().zip(right.chars()).take_while(|&(l, r)| l == r).count()
+/*
+FIXME: waiting for https://github.com/rust-lang/rust/pull/31700
+fn partial_match(pre: &str, name: &str) -> usize {
+    // skip(1) to ensure that the prefix never takes the whole variant name
+    pre.chars().zip(name.chars().rev().skip(1).rev()).take_while(|&(l, r)| l == r).count()
 }
 
-fn partial_rmatch(left: &str, right: &str) -> usize {
-    left.chars().rev().zip(right.chars().rev()).take_while(|&(l, r)| l == r).count()
+fn partial_rmatch(post: &str, name: &str) -> usize {
+    // skip(1) to ensure that the postfix never takes the whole variant name
+    post.chars().rev().zip(name.chars().skip(1).rev()).take_while(|&(l, r)| l == r).count()
+}*/
+
+fn partial_match(pre: &str, name: &str) -> usize {
```

Comment:
We actually could fold the camel_case methods into this. All we need to do is ensure that the next character is uppercase and at least one of the current ones is not.